### PR TITLE
Human vs random computer 

### PR DESCRIPTION
**Pull request by Himalee**
Given: I am a player
When: TTT starts
Then: I can choose to play human vs human or human vs random computer player
And: The game is set up correctly

### CODE REVIEW

File: src/components/game/Game.jsx [hunk 1]
```diff
@@ -3,22 +3,35 @@ import {Board} from '../board/Board';
...
+      case gameMode.HUMANVSRANDOM:
+        if (isMoveAvailable(grid, index) && !isGameOver(grid)) {
+          grid[index] = gameHelper.determineMark(this.state.grid);
+          grid[getMove(grid)] = gameHelper.determineMark(grid);
+        }
+        break;
+      default:
+        this.setState({grid: grid});
     }
     this.setState({grid: grid});
```
Comment (ssunday) [inline src/components/game/Game.jsx]: grid state set twice?

File: src/components/ticTacToe/TicTacToe.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  handleClick(e) {
+    const chosenGameMode = e.target.id;
+    this.setState({gameMode: chosenGameMode});
+  }
+
+  setUpGame() {
+    let game;
+    if (this.state.gameMode === gameMode.HUMANVSHUMAN) {
+      game = this.humanVsHumanGame();
```
Comment (ssunday) [inline src/components/ticTacToe/TicTacToe.jsx]: Why not return within the clauses?

File: src/components/ticTacToe/TicTacToe.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import React, {Component} from 'react';
+import {Game} from '../../components/game/Game';
+import {EMPTY} from '../../cellValue';
```
Comment (ssunday) [inline src/components/ticTacToe/TicTacToe.jsx]: spaces around brackets

File: src/components/ticTacToe/TicTacToe.jsx (hunk 2 above)
Comment (Himalee) [reply src/components/ticTacToe/TicTacToe.jsx]: Done!

File: src/components/game/Game.jsx (hunk 1 above)
Comment (Himalee) [reply src/components/game/Game.jsx]: I have removed the default case in the switch statement

File: src/components/ticTacToe/TicTacToe.jsx [hunk 4]
```diff
@@ -12,28 +12,23 @@ export class TicTacToe extends Component {
...
+  createGame(gameMode) {
+    return <Game board={Array(9).fill(EMPTY)} gameMode={gameMode} />;
+  }
+
   setUpGame() {
-    if (this.state.gameMode === gameMode.HUMANVSHUMAN) {
-      return this.humanVsHumanGame();
-    } else if (this.state.gameMode === gameMode.HUMANVSRANDOM) {
-      return this.humanVsRandomGame();
+    switch (this.state.gameMode) {
```
Comment (ssunday) [inline src/components/ticTacToe/TicTacToe.jsx]: woot

File: src/components/ticTacToe/TicTacToe.jsx [hunk 5]
```diff
@@ -8,40 +8,33 @@ export class TicTacToe extends Component {
...
+          <div>
+            <button id={gameMode.HUMANVSHUMAN} onClick={this.handleClick}>
+              Human
+            </button>
+            <button id={gameMode.HUMANVSRANDOM} onClick={this.handleClick}>
+              Random
+            </button>
+          </div>
+        ) : null}
+        {this.state.gameModeChosen ? (
```
Comment (ssunday) [inline src/components/ticTacToe/TicTacToe.jsx]: can't you do && instead of ternary?

**Pull request merged**